packages/block-library/src/footnotes/format.js

```diff
@@ -44,13 +48,26 @@ export const format = {
 		const registry = useRegistry();
 		const {
 			getSelectedBlockClientId,
+			getBlocks,
 			getBlockRootClientId,
 			getBlockName,
-			getBlocks,
+			getBlockParentsByBlockName,
 		} = useSelect( blockEditorStore );
 		const footnotesBlockType = useSelect( ( select ) =>
 			select( blocksStore ).getBlockType( name )
 		);
+		const hasParentCoreBlocks = useSelect( ( select ) => {
```

Comment:
Nit: `hasParentCodeBlocks` could sound like it means whether or not there is any core block as a parent. Would something like `isBlockWithinPattern` be more direct?

Reply:
Fair point. Will change quickly. Thanks for re-testing this (and for committing the improvement)